src/app/components/home/home.component.ts

```diff
@@ -8,17 +14,50 @@ import { Component, OnInit } from '@angular/core';
   templateUrl: './home.component.html',
   styleUrls: ['./home.component.scss']
 })
-export class HomeComponent implements OnInit {
+export class HomeComponent implements OnInit, OnDestroy {
 
-  /**
-   * Constructor.
-   */
-  constructor() {}
+  public user = {};
+  public challengeList = [];
+  authServiceSubscription: any;
+  constructor(private apiService: ApiService, private endpointService: EndpointsService, private authService: AuthService,
+              private globalService: GlobalService, private router: Router) { }
 
-  title = 'EvalAI|Home';
+  ngOnInit() {
+    this.init();
+    this.getChallenge();
+  }
 
-  /**
-   * Component on initialized.
-   */
-  ngOnInit() {}
+  init() {
+    const logInStatus = this.authService.isLoggedIn();
+    this.authServiceSubscription = this.authService.change.subscribe((authState) => {
+      if (authState.isLoggedIn) {
+        this.user = authState;
+      }
+    });
+  }
+
+  getChallenge() {
+    this.apiService.getUrl(this.endpointService.featuredChallengesURL()).subscribe(
+      response => {
+        this.challengeList = response.results;
+      },
+      err => { this.globalService.handleApiError(err); },
+      () => {}
+    );
+
+  }
+
+  hostChallenge() {
+    if (this.authService.isAuth) {
+      this.router.navigate(['/challenge-create']);
+    } else {
+      this.router.navigate(['/auth/login']);
```

Comment:
Don't hard-code the URL's; like in previous PR's, define them as constants above.

Reply:
done